stan/math/prim/prob/chi_square_lpdf.hpp

```diff
@@ -61,64 +69,59 @@ return_type_t<T_y, T_dof> chi_square_lpdf(const T_y& y, const T_dof& nu) {
     }
   }
 
-  if (!include_summand<propto, T_y, T_dof>::value) {
-    return 0.0;
-  }
-
   using std::log;
 
   VectorBuilder<include_summand<propto, T_y, T_dof>::value, T_partials_return,
                 T_y>
-      log_y(size(y));
-  for (size_t i = 0; i < stan::math::size(y); i++) {
-    log_y[i] = log(value_of(y_vec[i]));
-  }
-
+      log_y(size_y);
   VectorBuilder<include_summand<propto, T_y>::value, T_partials_return, T_y>
-      inv_y(size(y));
-  for (size_t i = 0; i < stan::math::size(y); i++) {
```

Comment:
Looks like the template logic changed. Now we're checking T_dof. Was there a bug?

Reply:
Earlier we had:
```
VectorBuilder<include_summand<propto, T_y, T_dof>::value, T_partials_return, T_y>
    log_y(size(y));
for (size_t i = 0; i < stan::math::size(y); i++) {
  log_y[i] = log(value_of(y_vec[i]));
}
```
With this change I'm doing the for loop if `include_summand<propto, T_y, T_dof>::value` is true, as per that VectorBuilder definition. But it doesn't matter, as if `include_summand<propto, T_y, T_dof>::value` is true, we've already returned 0 before reaching there. So neither the old version not this one is buggy, but it can be further simplified.